lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts

```diff
@@ -170,6 +173,130 @@ describe("Acquires a token successfully via an IMDS Managed Identity", () => {
         });
     });
 
+    describe("Linear Retry Policy", () => {
+        describe("User Assigned", () => {
+            let managedIdentityApplication: ManagedIdentityApplication;
+            beforeEach(() => {
+                managedIdentityApplication = new ManagedIdentityApplication(
+                    userAssignedClientIdConfig
+                );
+            });
+
+            test("returns a 500 error response from the network request, just the first time", async () => {
+                expect(ManagedIdentityTestUtils.isIMDS()).toBe(true);
+
+                const sendGetRequestAsyncSpy: jest.SpyInstance = jest
+                    .spyOn(networkClient, <any>"sendGetRequestAsync")
+                    // override the networkClient's sendGetRequestAsync method to return a 500.
+                    // after this override, original functionality will be restored
+                    // and the network request will complete successfully
+                    .mockReturnValueOnce(
+                        managedIdentityNetworkErrorClient.sendGetRequestForRetryAsync()
+                    );
+
+                const networkManagedIdentityResult: AuthenticationResult =
+                    await managedIdentityApplication.acquireToken(
+                        managedIdentityRequestParams
+                    );
+
+                expect(sendGetRequestAsyncSpy).toHaveBeenCalledTimes(2);
+                expect(networkManagedIdentityResult.accessToken).toEqual(
+                    DEFAULT_USER_SYSTEM_ASSIGNED_MANAGED_IDENTITY_AUTHENTICATION_RESULT.accessToken
+                );
+
+                jest.restoreAllMocks();
+            });
+
+            test("returns a 500 error response from the network request permanently", async () => {
+                expect(ManagedIdentityTestUtils.isIMDS()).toBe(true);
+
+                const sendGetRequestAsyncSpy: jest.SpyInstance = jest
+                    .spyOn(networkClient, <any>"sendGetRequestAsync")
+                    // permanently override the networkClient's sendGetRequestAsync method to return a 500.
+                    .mockReturnValue(
+                        managedIdentityNetworkErrorClient.sendGetRequestForRetryAsync()
+                    );
+
+                let serverError: ServerError = new ServerError();
+                try {
+                    await managedIdentityApplication.acquireToken(
+                        managedIdentityRequestParams
+                    );
+                } catch (e) {
+                    serverError = e as ServerError;
+                }
+
+                expect(serverError.errorCode).toEqual(
+                    MANAGED_IDENTITY_TOKEN_RETRIEVAL_ERROR
+                );
+                expect(sendGetRequestAsyncSpy).toHaveBeenCalledTimes(2);
+
+                jest.restoreAllMocks();
+            });
+        });
+
+        describe("System Assigned", () => {
```

Comment:
I don't think these tests add anything new. Leave them since you coded them up, but the code path is the same for UAMI and SAMI in relation to HTTP